### DISCUSSION TITLE
I want to make tailwindcss.config.ts follow tsconfig compilerOptions

### DISCUSSION THREAD

**Discussion by maeng2418** [tailwindlabs/tailwindcss#12037] [Help]
This is working code.
```
import plugins from './src/plugins/index'

export default {
  content: ['./src/**/*.{js,ts,jsx,tsx,mdx}'],
  plugins: [plugins.typographyPlugin],
}
```

This is not working code.
```
import plugins from '@plugins/index'

export default {
  content: ['./src/**/*.{js,ts,jsx,tsx,mdx}'],
  plugins: [plugins.typographyPlugin],
}
```

I wrote tsconfig.json like this.
```
{
  "compilerOptions": {
    "@/*": ["src/*"],
    "paths": {
      "@plugins/*": ["src/plugins/*"]
    }
  }
}
```
Even I checked if this path is wrong. But, it was working in other... (eg. app.tsx, and other component file level)

I wrote the file from './src/plugins/index.ts'
```
import typographyPlugin from '@tailwindcss/typography'

export default { typographyPlugin }
```

<img width="1819" alt="image" src="https://github.com/tailwindlabs/tailwindcss/assets/46251629/aa722d61-7766-4c98-a4d4-9d856d4b6fc6">

_Originally posted by @maeng2418 in https://github.com/tailwindlabs/tailwindcss/discussions/12034_

**Comment by maeng2418** [comment]
I'm not sure that tailwindcss.config.ts doesn't follow "compilerOptions" from tsconfig.json but it follows "jiti".
But, I think tailwindcss.config file should be compiled with tsconfig

**Comment by RobinMalfait** [comment]
Hey!

Internally we use `jiti` which allows us to use TypeScript for the Tailwind config. 

However, if you want to use a custom setup with path aliases, then I would recommend to compile your `tailwind.config.ts` file to a normal `tailwind.config.js` file instead.

Hope this helps!
